### ISSUE TITLE
Kokkos many task parallelism for GPUs

### ISSUE THREAD

**Issue by keitat** [kokkos/kokkos#1670]
I am working on a program that creates many tasks (but same computation ) on GPUs, and I wish each task executes small GPU kernels written with parallel_for.   I am curious if this is possible with the current version of Kokkos.

**Comment by mhoemmen** [comment]
@keitat Yes, Kokkos implements tasks that run in parallel on the GPU.  Check out [Tacho in Trilinos](https://github.com/trilinos/Trilinos/tree/master/packages/shylu/shylu_node/tacho) for an example.

**Comment by ibaned** [comment]
But right now its not possible for tasks to execute `parallel_for` depending on the amount of work in each task, this might be better suited to hierarchical parallelism.

**Comment by ibaned** [comment]
It may also be doable using streams, for which the new issue is #1723. For now I'll close this one, please comment if you'd like to reopen.
